### PR TITLE
Replace hard-coded Spark settings with config variables

### PR DESCRIPTION
**Pull request by mxm**
While playing around with the spark-pr-dashboard, I discovered that the variables in the settings file are not always respected. That led to a pretty nasty outcome :) Now, I would like to fix this such that other people, who blindly run the dashboard don't experience anything similar.

Other than that, kudos to everyone who helped putting this together. It's a nice piece of software.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ install:
 sudo: false
 script:
   - pep8
+  - pip install Flask Flask-Cache
```
Comment (JoshRosen) [inline .travis.yml]: This should use `requirements.txt` instead.

File: sparkprs/utils.py [hunk 2]
```diff
@@ -71,19 +72,21 @@ def parse_pr_title(pr_title):
...
     #  search the entire pull request title for JIRAs:
-    jiras = [int(x) for x in re.findall(r"SPARK-(\d+)", pr_title, re.I)]
+    jiras = [int(x) for x in re.findall(r"%s-(\d+)" % app.config['JIRA_PROJECT'], pr_title, re.I)]
     # Remove JIRAs from the metadata:
-    metadata_without_jiras = re.sub(r"\[?SPARK-\d+\]?", "", metadata, flags=re.I)
-    # Remove certain tags, since they're generally noise once the PRs are categorized:
-    tags_to_remove = ["MLLIB", "ML", "CORE", "PYSPARK", "SQL", "STREAMING", "YARN", "GRAPHX"]
+    metadata_without_jiras = re.sub(r"\[?%s-\d+\]?" % app.config['JIRA_PROJECT'],
+                                    "", metadata, flags=re.I)
+    # Remove certain tags, since they're generally noise once the PRs are
```
Comment (JoshRosen) [inline sparkprs/utils.py]: This lost a word from the original comment: "categorized"

Comment (mxm) [inline sparkprs/utils.py]: Thanks.

File: .travis.yml (hunk 1 above)
Comment (mxm) [inline .travis.yml]: That's right.

Comment (mxm) [inline .travis.yml]: I've included the requirements in the Travis file now.

**Comment by JoshRosen** [comment]
Thanks for working on this. I've tested out these changes and they worked fine, so I've merged this. Note that there are likely other parts of this project that are highly Spark-specific, so additional changes will likely be necessary if you want to use all of this tool's features with non-Spark projects.

**Comment by mxm** [comment]
Thanks for merging this! I realized that the Jenkins and QA bot integration is still Spark specific but I wanted the basic functionality to work independently of the project first. I might open a follow-up pull request. In the meantime, thanks for getting back to me so quickly.

**Pull request closed without merging**